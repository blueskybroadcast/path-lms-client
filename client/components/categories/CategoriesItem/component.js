import React from 'react';
import PropTypes from 'prop-types';

import CategoryFormModal from '../CategoryFormModal';

class CategoriesItem extends React.Component {
  state = {
    showEditModal: false
  }

  handleShow = () => {
    this.setState({
      showEditModal: true
    });
  }

  handleClose = () => {
    this.setState({
      showEditModal: false
    });
  }

  handleCategoryEdit = (category) => {
    const { editCategory, id } = this.props;

    editCategory(id, category);
  }

  handleCategoryDelete = () => {
    const confirmDelete = confirm('Are you sure you want to delete this category?');

    if (confirmDelete) {
      const { deleteCategory, id } = this.props;
      deleteCategory(id);
    }
  }

  render() {
    const {
      name, parent, hidden, level
    } = this.props;
    const { showEditModal } = this.state;

    return (
      <li className={`level-${level}`}>
        <span className="category">
          { level !== 0 && (
            <>
              <i className="icon icon-level-up icon-rotated-90" />
              &nbsp;
            </>
          )}
          <span className={level > 0 || hidden ? 'italic' : 'bold-weight'}>{name}</span>
        </span>
        <div className="actions">
          <a
            className="edit"
            href="#edit-category"
            onClick={this.handleShow}
          >
            <i className="icon icon-cog" />
            &nbsp;Edit&nbsp;
          </a>
          { showEditModal && (
            <CategoryFormModal
              show={showEditModal}
              handleClose={this.handleClose}
              handleSubmit={this.handleCategoryEdit}
              mode="edit"
              name={name}
              parentId={parent.id}
              hidden={hidden}
            />
          )}
          <a
            onClick={this.handleCategoryDelete}
            className="delete"
            href="#"
          >
            <i className="icon icon-times" />
            &nbsp;Delete
          </a>
        </div>
      </li>
    );
  }
}

CategoriesItem.propTypes = {
  name: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  parent: PropTypes.shape({
    id: PropTypes.string,
    type: PropTypes.string
  }),
  hidden: PropTypes.bool.isRequired,
  editCategory: PropTypes.func.isRequired,
  deleteCategory: PropTypes.func.isRequired,
  level: PropTypes.number.isRequired
};

CategoriesItem.defaultProps = {
  parent: null
};

export default CategoriesItem;
