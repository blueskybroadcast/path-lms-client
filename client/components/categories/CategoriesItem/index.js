import CategoriesItem from './container';

export default CategoriesItem;
