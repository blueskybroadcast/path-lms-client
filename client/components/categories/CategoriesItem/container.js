import { connect } from 'react-redux';

import CategoriesItem from './component';

import { editCategory, deleteCategory } from '../../../actions/categories';

export default connect(null, { editCategory, deleteCategory })(CategoriesItem);
