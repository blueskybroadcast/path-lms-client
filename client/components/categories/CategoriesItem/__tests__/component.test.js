import React from 'react';
import { shallow } from 'enzyme';
import CategoriesItem from '../component';

test('should render CategoriesItem correctly', () => {
  const props = {
    name: 'Category',
    id: '1',
    parent: {
      id: '8',
      type: 'type'
    },
    hidden: false,
    editCategory: jest.fn(),
    deleteCategory: jest.fn(),
    level: 1
  };

  const wrapper = shallow(<CategoriesItem {...props} />);
  expect(wrapper).toMatchSnapshot();
});
