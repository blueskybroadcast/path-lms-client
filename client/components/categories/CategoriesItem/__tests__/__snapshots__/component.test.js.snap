// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CategoriesItem correctly 1`] = `
<li
  className="level-1"
>
  <span
    className="category"
  >
    <i
      className="icon icon-level-up icon-rotated-90"
    />
     
    <span
      className="italic"
    >
      Category
    </span>
  </span>
  <div
    className="actions"
  >
    <a
      className="edit"
      href="#edit-category"
      onClick={[Function]}
    >
      <i
        className="icon icon-cog"
      />
       Edit 
    </a>
    <a
      className="delete"
      href="#"
      onClick={[Function]}
    >
      <i
        className="icon icon-times"
      />
       Delete
    </a>
  </div>
</li>
`;
